<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>time_calculate_window</class>
 <widget class="QMainWindow" name="time_calculate_window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>748</width>
    <height>469</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <spacer name="verticalSpacer_10">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QComboBox" name="comboBox">
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QComboBox {
    border: 1px solid gray;
    border-radius: 3px;
    padding: 1px 18px 1px 3px;
    min-width: 6em;
}

QComboBox:editable {
    background: white;
}

QComboBox:!editable, QComboBox::drop-down:editable {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
}

/* QComboBox gets the &quot;on&quot; state when the popup is open */
QComboBox:!editable:on, QComboBox::drop-down:editable:on {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #D3D3D3, stop: 0.4 #D8D8D8,
                                stop: 0.5 #DDDDDD, stop: 1.0 #E1E1E1);
}

QComboBox:on { /* shift the text when the popup opens */
    padding-top: 3px;
    padding-left: 4px;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 15px;

    border-left-width: 1px;
    border-left-color: darkgray;
    border-left-style: solid; /* just a single line */
    border-top-right-radius: 3px; /* same radius as the QComboBox */
    border-bottom-right-radius: 3px;
}

QComboBox::down-arrow {
    image: url(:/new/prefix1/img/icons8-sort-down-30.png);
width: 70%;
}

QComboBox::down-arrow:on { /* shift the arrow when popup is open */
    top: 1px;
    left: 1px;
}
QComboBox QAbstractItemView {
    border: 2px solid darkgray;
    selection-background-color: lightgray;

color: white;
    background:  #ABABAB;
}</string>
        </property>
        <item>
         <property name="text">
          <string>Bubble sorting</string>
         </property>
         <property name="icon">
          <iconset resource="../../yahzwhatisit.qrc">
           <normaloff>:/new/prefix1/img/B.png</normaloff>:/new/prefix1/img/B.png</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Merge sorting</string>
         </property>
         <property name="icon">
          <iconset resource="../../yahzwhatisit.qrc">
           <normaloff>:/new/prefix1/img/M.jpg</normaloff>:/new/prefix1/img/M.jpg</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Quick sorting</string>
         </property>
         <property name="icon">
          <iconset resource="../../yahzwhatisit.qrc">
           <normaloff>:/new/prefix1/img/Q.png</normaloff>:/new/prefix1/img/Q.png</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Tree sorting</string>
         </property>
         <property name="icon">
          <iconset resource="../../yahzwhatisit.qrc">
           <normaloff>:/new/prefix1/img/Creative-Tail-Halloween-tree.svg.png</normaloff>:/new/prefix1/img/Creative-Tail-Halloween-tree.svg.png</iconset>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Radix sorting</string>
         </property>
         <property name="icon">
          <iconset resource="../../yahzwhatisit.qrc">
           <normaloff>:/new/prefix1/img/depositphotos_9101833-stock-photo-grass-letter-r.jpg</normaloff>:/new/prefix1/img/depositphotos_9101833-stock-photo-grass-letter-r.jpg</iconset>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_11">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="situation_arg1">
        <property name="styleSheet">
         <string notr="true">color:  skyblue;
</string>
        </property>
        <property name="text">
         <string>Enter count elements</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="elements_count">
        <property name="maximumSize">
         <size>
          <width>122</width>
          <height>22</height>
         </size>
        </property>
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">QSpinBox {
    padding-right: 15px; /* make room for the arrows */
/*    border-image: url(:/images/frame.png); */
    border-width: 3;
color: white;
 selection-background-color: orange;
  background-color: #3B6582;
}

QSpinBox::up-button {
    subcontrol-origin: border;
    subcontrol-position: top right; /* position at the top right corner */

    width: 16px; /* 16 + 2*1px border-width = 15px padding + 3px parent border */
    border-image: url(:/images/spinup.png) 1;
    border-width: 1px;
}

QSpinBox::up-button:hover {
    border-image: url(:/images/spinup_hover.png) 1;
}

QSpinBox::up-button:pressed {
    border-image: url(:/images/spinup_pressed.png) 1;
}

QSpinBox::up-arrow {
    image: url(:/images/up_arrow.png);
    width: 7px;
    height: 7px;
}

QSpinBox::up-arrow:disabled, QSpinBox::up-arrow:off { /* off state when value is max */
   image: url(:/images/up_arrow_disabled.png);
}

QSpinBox::down-button {
    subcontrol-origin: border;
    subcontrol-position: bottom right; /* position at bottom right corner */

    width: 16px;
    border-image: url(:/images/spindown.png) 1;
    border-width: 1px;
    border-top-width: 0;
}

QSpinBox::down-button:hover {
    border-image: url(:/images/spindown_hover.png) 1;
}

QSpinBox::down-button:pressed {
    border-image: url(:/images/spindown_pressed.png) 1;
}

QSpinBox::down-arrow {
    image: url(:/images/down_arrow.png);
    width: 7px;
    height: 7px;
}

QSpinBox::down-arrow:disabled,
QSpinBox::down-arrow:off { /* off state when value in min */
   image: url(:/images/down_arrow_disabled.png);
}
</string>
        </property>
        <property name="maximum">
         <number>999999999</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="consider">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton {
  background-color: #3B6582;
  color: white; 
 width:  100px;
 height: 25px;
  border: 1px solid gray;
}


QPushButton:hover {
    background-color: #E1E8EE;
color: black;
}
QPushButton:pressed {  background-color rgb(220, 116, 45) 
}</string>
        </property>
        <property name="text">
         <string>Сalculate</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <spacer name="verticalSpacer_6">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <spacer name="verticalSpacer_7">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="styleSheet">
         <string notr="true">color:  skyblue;
font-size: 9px;</string>
        </property>
        <property name="text">
         <string>Time comlpexity</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="maximumSize">
         <size>
          <width>50</width>
          <height>15</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QProgressBar {
    border: 2px solid grey;
   background-color: rgba(29, 55, 91, 10);
    border-radius: 5px;
    text-align: center;
    color: red;
}
QProgressBar::chunk {
    background-color: #05B8CC;  /*  */
    width: 20px;
}
</string>
        </property>
        <property name="value">
         <number>24</number>
        </property>
        <property name="textVisible">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="2">
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="sizeConstraint">
       <enum>QLayout::SetFixedSize</enum>
      </property>
      <item row="2" column="1">
       <spacer name="verticalSpacer_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QLabel" name="label_4">
        <property name="maximumSize">
         <size>
          <width>210</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">
color:  skyblue;
font-size: 18px;    </string>
        </property>
        <property name="text">
         <string>Time is a great treasure</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="0">
       <spacer name="verticalSpacer_8">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="QLabel" name="quote">
        <property name="maximumSize">
         <size>
          <width>250</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;right&quot;&gt;&lt;span style=&quot; font-size:10pt; font-style:italic; color:#87ceeb;&quot;&gt;Lost time is never found again.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <spacer name="verticalSpacer_5">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="4" column="0" colspan="2">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="sizeConstraint">
         <enum>QLayout::SetFixedSize</enum>
        </property>
        <item>
         <widget class="QLabel" name="label">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>10</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">
                              /*&quot;background-color: rgba(176, 196, 222, 90);&quot;*/
color:  skyblue;
font-size: 22px;
</string>
          </property>
          <property name="text">
           <string>Run Time</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLCDNumber" name="lcdNumber">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>37</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="statusTip">
           <string/>
          </property>
          <property name="layoutDirection">
           <enum>Qt::RightToLeft</enum>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">color: skyblue;
/*margin-right: 10%;*/
   font-size: 18px;
                                  /*&quot;background-color: rgba(176, 196, 222, 90);&quot;*/ </string>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="lineWidth">
           <number>0</number>
          </property>
          <property name="digitCount">
           <number>6</number>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::Flat</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="1">
       <spacer name="verticalSpacer_9">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>748</width>
     <height>19</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QMenu {
    background-color: #ABABAB; /* sets background of the menu */

    border: 1px solid black;
}

QMenu::item {
    /* sets background of menu item. set this to something non-transparent
        if you want menu color and menu item color to be different */
    background-color: transparent;
}

QMenu::item:selected { /* when user selects item using mouse or keyboard */
    background-color: steelblue;
}
QMenuBar {
    background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                      stop:0 lightgray, stop:1 darkgray);
color: steelblue;
}

QMenuBar::item {
    spacing: 3px; /* spacing between menu bar items */
    padding: 1px 4px;
    background: transparent;
    border-radius: 4px;
}

QMenuBar::item:selected { /* when selected using mouse or keyboard */
    background: #a8a8a8;
}

QMenuBar::item:pressed {
    background: #888888;
}</string>
   </property>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <widget class="QMenu" name="menuLanguage">
     <property name="title">
      <string>Language</string>
     </property>
     <property name="icon">
      <iconset resource="../../yahzwhatisit.qrc">
       <normaloff>:/new/prefix1/img/Hammer_and_sickle_red_on_transparent.svg_-1024x1024.png</normaloff>:/new/prefix1/img/Hammer_and_sickle_red_on_transparent.svg_-1024x1024.png</iconset>
     </property>
     <addaction name="actionEnglish"/>
     <addaction name="actionRussian"/>
     <addaction name="actionOthers"/>
    </widget>
    <addaction name="menuLanguage"/>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuAdd_sort">
    <property name="title">
     <string>Add sort</string>
    </property>
    <addaction name="actionLet_s_try"/>
   </widget>
   <addaction name="menuSettings"/>
   <addaction name="menuAdd_sort"/>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="styleSheet">
    <string notr="true">color:  skyblue;</string>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="styleSheet">
    <string notr="true">    background-color: #ABABAB;



QToolBar::item {
    background: red;
}
QToolBar::item:selected { /* when user selects item using mouse or keyboard */
    background-color: #ABABAB;
}
</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonIconOnly</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionBack"/>
   <addaction name="actionRefresh"/>
  </widget>
  <action name="actionEnglish">
   <property name="icon">
    <iconset resource="../../yahzwhatisit.qrc">
     <normaloff>:/new/prefix1/img/usa.jpg</normaloff>:/new/prefix1/img/usa.jpg</iconset>
   </property>
   <property name="text">
    <string>English</string>
   </property>
  </action>
  <action name="actionRussian">
   <property name="icon">
    <iconset>
     <normaloff>:/new/prefix1/img/rus.png</normaloff>:/new/prefix1/img/rus.png</iconset>
   </property>
   <property name="text">
    <string>Russian</string>
   </property>
  </action>
  <action name="actionOthers">
   <property name="text">
    <string>Others</string>
   </property>
  </action>
  <action name="actionBack">
   <property name="icon">
    <iconset resource="../../yahzwhatisit.qrc">
     <normaloff>:/new/prefix1/img/Arrow_left.png</normaloff>:/new/prefix1/img/Arrow_left.png</iconset>
   </property>
   <property name="text">
    <string>Back</string>
   </property>
   <property name="toolTip">
    <string>Back</string>
   </property>
  </action>
  <action name="actionRefresh">
   <property name="icon">
    <iconset resource="../../yahzwhatisit.qrc">
     <normaloff>:/new/prefix1/img/2183585b41e7.png</normaloff>:/new/prefix1/img/2183585b41e7.png</iconset>
   </property>
   <property name="text">
    <string>Refresh</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset resource="../../yahzwhatisit.qrc">
     <normaloff>:/new/prefix1/img/Manager_Icon_thumb.png</normaloff>:/new/prefix1/img/Manager_Icon_thumb.png</iconset>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionLet_s_try">
   <property name="text">
    <string>Let's try!</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../../yahzwhatisit.qrc"/>
 </resources>
 <connections/>
</ui>
